testsuite/testsuite-undertow/src/test/java/org/wildfly/swarm/undertow/HTTPSCustomizerSelfSignedCertificateTest.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2015-2016 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.swarm.undertow;
+
+import java.net.URL;
+
+import javax.net.ssl.SSLHandshakeException;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.swarm.Swarm;
+import org.wildfly.swarm.arquillian.CreateSwarm;
+import org.wildfly.swarm.management.ManagementFraction;
+import org.wildfly.swarm.spi.api.SwarmProperties;
+
+/**
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+@RunWith(Arquillian.class)
+public class HTTPSCustomizerSelfSignedCertificateTest {
+
+    @Deployment(testable = false)
+    public static Archive createDeployment() {
+        return ShrinkWrap.create(WARArchive.class, "test.war")
+                .add(EmptyAsset.INSTANCE, "index.html");
+    }
+
+    @CreateSwarm
+    public static Swarm newSwarm() throws Exception {
+        System.setProperty(SwarmProperties.HTTPS_GENERATE_SELF_SIGNED_CERTIFICATE, "true");
+        return new Swarm()
+                .fraction(UndertowFraction.createDefaultFraction())
+                .fraction(ManagementFraction.createDefaultFraction());
+    }
+
+    @Test(expected = SSLHandshakeException.class)
```

Comment:
I'm not an expert on SSL, but should this test throw a handshake exception if a self signed cert is installed?